### DISCUSSION TITLE
How to get top level object where nested relation occurs at least once

### DISCUSSION THREAD

**Discussion by paymog** [prisma/prisma#22745] [Q&A]
### Question

I have the following query

```js
  const projectsAndSubgraphs = await prisma.project.findMany({
    select: {
      id: true,
      billing_subscription_id: true,
      subgraph_deployments: {
        select: {
          subgraph_deployment_id: true,
        },
        where: {
          subgraph_deployment_id: {
            notIn: communitySubgraphs,
          },
          status: SubgraphStatus.Active,
        },
      },
      default_graph_node_cluster: {
        select: {
          name: true,
        },
      },
    },
    where: {
      billing_subscription_id: {
        startsWith: "sub_",
      },
    },
  });
```

I find that this will return projects that have 0 associated subgraphs. Is there a way to rewrite this query to only return projects which have at least 1 subgraph? This is related to #22744 because `JOIN`s are not being used.

I'm using prisma 5.2.0.

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
model SubgraphDeployment {
  id                       String           @id @default(cuid())
  project_id               String
  project                  Project          @relation(fields: [project_id], references: [id])
  graph_node_cluster_id    String
  graph_node_cluster       GraphNodeCluster @relation(fields: [graph_node_cluster_id], references: [id])
}

model GraphNodeCluster {
  id             String  @id @default(cuid())
  name           String
}

model Project {
  id                            String               @id @default(cuid())
  default_graph_node_cluster_id String
  default_graph_node_cluster    GraphNodeCluster     @relation(fields: [default_graph_node_cluster_id], references: [id])

  subgraph_deployments SubgraphDeployment[]

}
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment]
Hi @paymog 👋 

Thank you for raising this question.

Can you please share a minimal schema in order to understand how Projects and Subgraphs relate?

**Reply by paymog** [reply]
just updated the question with a slimmed down version of my prisma model

**Reply by ludralph** [reply]
Hi @paymog, you would need to modify your query to use `some` filter. See the docs on [filtering](https://www.prisma.io/docs/orm/prisma-client/queries/filtering-and-sorting#filtering)

```js
const projectsAndSubgraphs = await prisma.project.findMany({
    select: {
      id: true,
      billing_subscription_id: true,
      subgraph_deployments: {
        select: {
          subgraph_deployment_id: true,
        },
        where: {
          subgraph_deployment_id: {
            notIn: communitySubgraphs,
          },
          status: SubgraphStatus.Active,
        },
      },
      default_graph_node_cluster: {
        select: {
          name: true,
        },
      },
    },
    include:{
      subgraph_deployments: true
    },
    where: {
      billing_subscription_id: {
        startsWith: "sub_",
      },
      subgraph_deployments: {
        some: {
          // specify filter condition
        },
      },
    },
  });
```

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
